src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs

```diff
@@ -12,28 +12,36 @@ namespace Microsoft.NET.Build.Tasks
 {
     internal static class LockFileExtensions
     {
-        public static LockFileTarget GetTargetAndThrowIfNotFound(this LockFile lockFile, NuGetFramework framework, string runtime)
+        public static LockFileTarget GetTargetAndThrowIfNotFound(this LockFile lockFile, NuGetFramework framework, string runtime,
+            bool throwIfNotFound = true)
```

Comment:
The API shape here is getting weird. `AndThrowIfNotFound` vs. `bool throwIfNotFound`

Having trouble coming up with a better shape though. Feels like it should just be GetTarget with the parameters modifying the behavior, but that conflicts with real instance GetTarget that can return null.